scripts/destroy-dev-stack.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -o errexit -o pipefail
+
+source ./scripts/common.sh
+
+# This script destroys a dev stack in the `pulumi` organization.
+#
+# Usage:
+#
+#   ./scripts/destroy-dev-stack.sh                         # Destroys a dev stack. Prompts for stack name.
+#   ./scripts/destroy-dev-stack.sh stackname               # Same as above, but uses the passed-in argument for the stack name.
+#   STACK_NAME=stackname ./scripts/destroy-dev-stack.sh    # Same as above, but uses the STACK_NAME environment variable instead.
+
+stack_name="${STACK_NAME:=$1}"
+
+if [ -z "$stack_name" ]; then
+    echo
+    echo "This script assumes you're logged into Pulumi Cloud and that you belong"
+    echo "to the 'pulumi' organization."
+    echo
+    echo "Available stacks:"
+    echo
+    pulumi -C infrastructure stack ls | grep -v "production" | grep -v "staging" | grep -v "testing"
+    echo
+    read -p "Which stack do you want to destroy? " stack_name
+fi
+
+if [[ "$stack_name" == *production* || "$stack_name" == *staging* || "$stack_name" == *testing* ]]; then
+    echo
+    echo "Refusing to deploy the $stack_name stack with this script. Exiting."
+    exit 0
+fi
+
+echo
+echo "Selecting the $stack_name stack..."
+pulumi -C infrastructure stack select "$stack_name"
+
+echo
+echo "Destroying ${stack_name}..."
+
+echo
+echo "Emptying the contents of potentially non-empty buckets..."
+echo
+stack_name_only=$(echo $stack_name | sed -e 's/\(pulumi\/\)//')
+
+pushd infrastructure
+    for resource_name in "website-logs-bucket" "uploads-bucket" "fallback-bucket"; do
```

Comment:
good catch! I actually think there is an option available now that we can set on the s3 bucket config to empty the bucket when we call `pulumi destroy` before it deletes the bucket. Though what we have now is good for protection though so we can't accidentally delete the buckets in prod. We could also probably get around this by setting the protected resource option though. We can follow up addressing this later.